lib/matplotlib/_mathtext.py

```diff
@@ -2599,3 +2604,23 @@ def auto_delim(self, s, loc, toks):
             # if "mid" in toks ... can be removed when requiring pyparsing 3.
             toks["mid"].asList() if "mid" in toks else [],
             toks["right"])
+
+    def substack(self, s, loc, toks):
+        state = self.get_state()
+        thickness = state.get_current_underline_thickness()
+        vlist = []
+
+        max_width = max(map(lambda c: c.width, toks[1:]))
+
+        for sub in toks[1:]:
+            cp = HCentered([sub])
+            cp.hpack(max_width, 'exactly')
+            vlist.append(cp)
+
+        vlist = [x for l in (vlist[i:i+1] +
+                             [Vbox(0, thickness * 2.0)] *
+                             (i < len(vlist) - 1)
+                             for i in range(0, len(vlist), 1)) for x in l]
```

Comment:
Took me an embarrassingly long time to understand!:sweat_smile: I think it separates each element of `vlist` with `Vbox(0, thickness * 2)`?

This is probably one of those instances I'd prioritise readability.
```
vlist_ = [None, Vbox(0, thickness * 2)] * len(vlist)
vlist_[::2] = vlist
vlist_.pop()
```
This should achieve this same thing. (Another option is to append the `Vbox`es in the `for` loop just above.)

However, I'd defer to @oscargus's opinion: what do you think?

Reply:
Maybe one could just do:

```
vlist = [val for pair in zip(vlist, [Vbox(0, thickness * 2)]*len(vlist)) for val in pair]
del vlist[-1]
```
?

(But, yes, I find double (and triple) list comprehensions quite hard to follow to start with.)